src/nvim/eval.c

```diff
@@ -5421,6 +5421,9 @@ static int list_join(garray_T *gap, list_T *l, char_u *sep, int echo_style, int
   garray_T join_ga;
   int retval;
 
+  if (l->lv_len < 1)
+    return OK;
```

Comment:
@bfredl Even if there are only two possible return types, I find that `true` and `false` can be a poor fit. Unless the function is named `is*()`, then I think that `OK` and `FAIL` are clearer.